app/assets/javascripts/discourse/app/controllers/tag-show.js

```diff
@@ -181,14 +181,7 @@ export default DiscoverySortableController.extend(
       this.tagNotification
         .update({ notification_level: notificationLevel })
         .then((response) => {
-          this.currentUser.set(
-            "muted_tag_ids",
-            this.currentUser.calculateMutedIds(
-              notificationLevel,
-              response.responseJson.tag_id,
-              "muted_tag_ids"
-            )
-          );
+          this.currentUser.setProperties(response.responseJson);
```

Comment:
This is a little bit risky. Look at the following case:

```
// json response
{
  foo: false
}

// currentUser model file
@computed
foo() {
  return true
}
```

It's very easy to erase a computed property, I think at very least we should be explicit on the keys we set here.

Reply:
Makes sense. I've updated the code to only set the keys we expect.